lib/ast/AST.cpp

```diff
@@ -43,11 +210,48 @@ void KOREDeclaration::addMetaSortVariable(KOREMetaSortVariable *SortVariable) {
   metaSortVariables.push_back(SortVariable);
 }
 
-
 void KOREAxiomDeclaration::addPattern(KOREPattern *Pattern) {
   pattern = Pattern;
 }
 
+static const std::string ASSOC = "assoc";
+static const std::string COMM = "comm";
+static const std::string IDEM = "idem";
+static const std::string UNIT = "unit";
+static const std::string FUNCTIONAL = "functional";
+static const std::string CONSTRUCTOR = "constructor";
+
+bool KOREAxiomDeclaration::isRequired() {
+  return !attributes.count(ASSOC) && !attributes.count(COMM) && !attributes.count(IDEM) && !attributes.count(UNIT) && !attributes.count(FUNCTIONAL) && !attributes.count(CONSTRUCTOR);
+}
+
+KOREPattern *KOREAxiomDeclaration::getRightHandSide() const {
+  if (auto top = dynamic_cast<KOREObjectCompositePattern *>(pattern)) {
```

Comment:
This seems really fragile, but I haven't thought of a better way to handle this, yet. Definitely replacing getName() with unique pointers to each builtin will clean this up some, but the static indices will still be here.